src/Repository.ts

```diff
@@ -46,8 +47,8 @@ export class MongoRepository<DOC, DTO = DOC> {
    * @returns {Promise<DOC | null>}
    * @memberof MongoRepository
    */
-  findById(id: string): Promise<DOC | null> {
-    return this.findOne({ _id: new ObjectID(id) });
+  findById(id: string): Promise<WithId<DOC> | null> {
```

Comment:
@jeanmurillo95 this remains unaddressed. Do you understand the feedback? 

On line 88, the `_Id` is replaced with `Id`. So the change to the return type here and on L83 are incorrect. (`tsc` isn't catching this because `toggleId()` returns `any`; fixing that would be a big improvement, but I don't recommend trying to fix that right now.)

Reply:
This is unresolved